### PR TITLE
feat: add support for getting DraftVersion03 response headers on the /check_and_report operation.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

File: limitador-server/docs/http_server_spec.json [hunk 1]
```diff
@@ -16,6 +16,10 @@
           "additionalProperties": {
             "type": "string"
           }
+        },
+        "response_headers": {
```
Comment (alexsnaps) [inline limitador-server/docs/http_server_spec.json:20]: Is this supposed to be camelCase? There seems to be a mix here already… `CheckAndReportInfo` vs `additionalProperties` (lowercase)… Don't know it there is a recommendation for either/or but camel & snake seems weird, no?

**Review by alexsnaps — APPROVED** [review]

File: limitador-server/docs/http_server_spec.json (hunk 1 above)
Comment (chirino) [reply limitador-server/docs/http_server_spec.json:20]: additionalProperties is not the name of a field... it's used to define the type of the key for dictionary types.  see: https://swagger.io/docs/specification/data-models/dictionaries/ 

The norm is CamelCase type names but snake case field names. Here is another field that is using sake case: https://github.com/Kuadrant/limitador/blob/b2695aec5e7ab128420a4a694ab968d29e840f3d/limitador-server/docs/http_server_spec.json#L80

**Comment by chirino** [comment]
rebased

**Comment by chirino** [comment]
rebased

**Pull request merged**